packages/core/src/command/consolidated/consolidated-filter-by-date.ts

```diff
@@ -0,0 +1,334 @@
+    if (!entry.resource) return false;
+    switch (entry.resource.resourceType) {
+      case "AllergyIntolerance":
+        return isAllergyIntoleranceWithinDateRange(entry.resource, range);
+      case "Communication":
+        return isCommunicationWithinDateRange(entry.resource, range);
+      case "Composition":
+        return isCompositionWithinDateRange(entry.resource, range);
+      case "Condition":
+        return isConditionWithinDateRange(entry.resource, range);
+      case "Consent":
+        return isConsentWithinDateRange(entry.resource, range);
+      case "Coverage":
+        return isCoverageWithinDateRange(entry.resource, range);
+      case "DiagnosticReport":
+        return isDiagnosticReportWithinDateRange(entry.resource, range);
+      case "DocumentReference":
+        return isDocumentReferenceWithinDateRange(entry.resource, range);
+      case "Encounter":
+        return isEncounterWithinDateRange(entry.resource, range);
+      case "FamilyMemberHistory":
+        return isFamilyMemberHistoryWithinDateRange(entry.resource, range);
+      case "Goal":
+        return isGoalWithinDateRange(entry.resource, range);
+      case "Immunization":
+        return isImmunizationWithinDateRange(entry.resource, range);
+      case "MedicationAdministration":
+        return isMedicationAdministrationWithinDateRange(entry.resource, range);
+      case "MedicationDispense":
+        return isMedicationDispenseWithinDateRange(entry.resource, range);
+      case "MedicationRequest":
+        return isMedicationRequestWithinDateRange(entry.resource, range);
+      case "MedicationStatement":
+        return isMedicationStatementWithinDateRange(entry.resource, range);
+      case "Observation":
+        return isObservationWithinDateRange(entry.resource, range);
+      case "Procedure":
+        return isProcedureWithinDateRange(entry.resource, range);
+      case "Provenance":
+        return isProvenanceWithinDateRange(entry.resource, range);
+      case "RelatedPerson":
+        return isRelatedPersonWithinDateRange(entry.resource, range);
+    }
+    return includeIfNoDate;
+  });
+  return {
+    ...bundle,
+    total: filteredEntries?.length ?? 0,
+    entry: filteredEntries ?? [],
+  };
+}
+
+function isAllergyIntoleranceWithinDateRange(
+  resource: AllergyIntolerance,
+  range: DateRange
+): boolean {
+  if (resource.onsetPeriod) {
+    const res = arePeriodsWithinRange([resource.onsetPeriod], range);
+    if (res !== undefined) return res;
+  }
+  if (resource.onsetRange) {
+    const res = areRangesWithinRange([resource.onsetRange], range);
+    if (res !== undefined) return res;
+  }
+  if (resource.onsetString) {
+    const res = isDateWithinDateRange(resource.onsetString, range);
+    if (res !== undefined) return res;
+  }
+  if (resource.onsetDateTime) {
+    const res = isDateWithinDateRange(resource.onsetDateTime, range);
+    if (res !== undefined) return res;
+  }
```

Comment:
missed resource.onsetAge of type Age

Reply:
Intentionally not using onsetAge because it requires a more complex solution that accounts for the patient. I already have something half-baked that I can send on the next PR if that's OK.

Reply:
yah thats good with me